gtars/src/uniwig/mod.rs

```diff
@@ -1,16 +1,722 @@
+    chromosome.ends.sort_unstable();
+    chromosome_vec.push(chromosome.clone());
+
+    println!("Reading Bed file complete.");
+
+    //chromosome_vec.sort_by_key(|c| c.chrom.clone());
+
+    return chromosome_vec
+
+}
+
+pub fn parse_bed_file(line: &str) -> Option<(String, i32, i32)> {
+    // TODO Eventually refactor all bed file parsing to a single shared function
+
+    let mut fields = line.split('\t');
+    // Get the first field which should be chromosome.
+    let ctg = fields.next()?;
+    // Parse 2nd and 3rd string as integers or return -1 if failure
+    let st = fields.next().and_then(|s| s.parse::<i32>().ok()).unwrap_or(-1);
+    let en = fields.next().and_then(|s| s.parse::<i32>().ok()).unwrap_or(-1);
+
+    // Original code had a remainder of the line, r, but it does not appear to have been used
+    // in any way
+
+    Some((ctg.parse().unwrap(), st, en))
+
+}
+
+
+pub fn run_uniwig(matches: &ArgMatches) {
+    //println!("I am running. Here are the arguments: {:?}", matches);
+
+    let combinedbedpath = matches
+        .get_one::<String>("bed")
+        .expect("combined bed path is required");
+
+    let chromsizerefpath = matches
+        .get_one::<String>("chromref")
+        .expect("chromref path path is required");
+
+    let bwfileheader = matches
+        .get_one::<String>("fileheader")
+        .expect("fileheader is required");
+
+    let smoothsize = matches
+        .get_one::<i32>("smoothsize")
+        .expect("smoothsize required");
+
+    let output_type = matches
+        .get_one::<String>("outputtype")
+        .expect("output type is required");
+
+
+    uniwig_main(*smoothsize, combinedbedpath, chromsizerefpath, bwfileheader, output_type)
+
+
+}
+
+fn clamped_start_position(start:i32, smoothsize: i32) -> i32{
+    // This is for ensuring that the start position for every wiggle file is at a minimum equal to `1`
+    std::cmp::max(1, start - smoothsize)
+
+}
+
+pub fn uniwig_main(smoothsize:i32, combinedbedpath: &str, _chromsizerefpath: &String, bwfileheader: &str, output_type: &str){
+    // Main Function
+
+    //println!("Hello from Uniwig main");
+
+    let stepsize = 1;
+
+    // Set up output file names
+
+    let mut file_names: [String; 3] = ["placeholder1".to_owned(), "placeholder2".to_owned(), "placeholder3".to_owned()];
+    let mut meta_data_file_names: [String; 3] = ["placeholder1".to_owned(), "placeholder2".to_owned(), "placeholder3".to_owned()];
+
+    // TODO determine potential file types
+    file_names[0] = format!("{}_{}.{}", bwfileheader, "start", output_type);
+    file_names[1] = format!("{}_{}.{}", bwfileheader, "end", output_type);
+    file_names[2] = format!("{}_{}.{}", bwfileheader, "core", output_type);
+
+    meta_data_file_names[0] = format!("{}{}.{}", bwfileheader, "start","meta");
+    meta_data_file_names[1] = format!("{}{}.{}", bwfileheader, "end","meta");
+    meta_data_file_names[2] = format!("{}{}.{}", bwfileheader, "core","meta");
```

Comment:
I might be misinterpreting, but is there a reason to store these in an array instead of just creating three new variables for them?

Reply:
Some of these items are almost 1 to 1 ports from the c code to help with ensuring I can pick up easily if I step away from it for a while. In this case, we could just create three new variables.

Reply:
I gotcha -- I think thats a valid approach for now haha